### PR TITLE
Update of dependencies

### PR DESCRIPTION
**Pull request by fbenz**

### CODE REVIEW

File: spring-auto-restdocs-core/pom.xml [hunk 1]
```diff
@@ -78,6 +78,10 @@
             <artifactId>javax.el-api</artifactId>
         </dependency>
         <dependency>
+            <groupId>org.glassfish.web</groupId>
+            <artifactId>javax.el</artifactId>
```
Comment (jmisur) [inline spring-auto-restdocs-core/pom.xml]: why is this necessary?

Comment (fbenz) [inline spring-auto-restdocs-core/pom.xml]: We had the el api as a dependency, but no implementation for it. Hibernate Validator started enforcing an implementation, see https://hibernate.atlassian.net/browse/HV-1131 and http://hibernate.org/validator/documentation/getting-started/

File: pom.xml [hunk 2]
```diff
@@ -220,28 +220,28 @@
...
                 <groupId>org.springframework.data</groupId>
                 <artifactId>spring-data-commons</artifactId>
-                <version>1.12.2.RELEASE</version>
+                <version>1.13.0.RELEASE</version>
             </dependency>
             <dependency>
                 <groupId>org.hibernate</groupId>
                 <artifactId>hibernate-validator</artifactId>
-                <version>5.2.3.Final</version>
+                <version>5.4.0.Final</version>
```
Comment (jmisur) [inline pom.xml]: 5.3.4.Final

File: pom.xml [hunk 3]
```diff
@@ -272,9 +272,14 @@
                 <version>2.2.5</version>
             </dependency>
             <dependency>
+                <groupId>org.glassfish.web</groupId>
+                <artifactId>javax.el</artifactId>
+                <version>2.2.6</version>
```
Comment (jmisur) [inline pom.xml]: spring-boot-starter-web includes tomcat-embed-el if tomcat is used (default).
I think some of the dependencies should not be included by default as they might conflict with whatever project uses.

**Comment by fbenz** [comment]
I switched to Hibernate Validator 5.3.4.Final and thus the same version as the latest Spring Boot version. An EL implementation is still required but I put the right scope on it: `test`. Just including `tomcat-embed-el` on its own does not work and thus I kept the Glassfish implementation.
